server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketSelectorPipelineAggregator.java

```diff
@@ -40,14 +38,12 @@
 
 public class BucketSelectorPipelineAggregator extends PipelineAggregator {
     private GapPolicy gapPolicy;
-
     private Script script;
-
     private Map<String, String> bucketsPathsMap;
 
-    public BucketSelectorPipelineAggregator(String name, Map<String, String> bucketsPathsMap, Script script, GapPolicy gapPolicy,
+    BucketSelectorPipelineAggregator(String name, Map<String, String> bucketsPathsMap, Script script, GapPolicy gapPolicy,
             Map<String, Object> metadata) {
-        super(name, bucketsPathsMap.values().toArray(new String[bucketsPathsMap.size()]), metadata);
+        super(name, bucketsPathsMap.values().toArray(new String[0]), metadata);
```

Comment:
I'm curious why this change is needed? My understanding was that if you initialise the array to the size of the map it avoids having to perform resizes which reinitialise the array inside the `toArray()` method.